test/verify/check-testlib

```diff
@@ -134,8 +129,15 @@ class TestRunTest(MachineCase):
         with open(test_file, 'r') as f:
             original_test = f.read()
 
-        self.addCleanup(writeFile, test_file, original_test)
-        writeFile(test_file, original_test.replace("class NoTest", "class Test"))
+        def write_file(file, content, mode=0o666):
+            # test files need to be executable, respect umask
+            fd = os.open(file, os.O_CREAT | os.O_TRUNC | os.O_WRONLY, mode=mode)
+            assert fd >= 0
+            os.write(fd, content.encode())
+            os.close(fd)
```

Comment:
If you wanted to be very very strict, though, you might decide you want to deal with the case where `write()` fails (`ENOSPC`?) and then you'll leak the `fd`...

I like to use `finally:` blocks when dealing with fds.

```python
try:
    os.write(fd, content.encode())
finally:
    os.close(fd)
```

Reply:
Good idea. I also strengthened the os.write() to assert a full write. It's "just" a test, so I think we can affort to be stricter than necessary, and not waffle around with retry loops and such.